src/main/scala/symsim/concrete/ConcreteQTable.scala

```diff
@@ -11,17 +11,17 @@ trait ConcreteQTable[State, ObservableState, Action]
 
   import agent.instances.*
 
-  def bestAction (q: Q) (s: State): Action =
-    val qs = q (agent.discretize (s)) map { _.swap }
+  def bestAction (q: Q) (s: ObservableState): Action =
+    val qs = q (s).map { _.swap }
     qs (qs.keys.max)
 
 
-  def chooseAction (q: Q) (s: State): Randomized[Action] =
+  def chooseAction (q: Q) (s: ObservableState): Randomized[Action] =
```

Comment:
Are you thinking for the algorithms that the action set is various for different states?
Otherwise, this does not make sense, because we should always call discretize function before choosingAction!